### DISCUSSION TITLE
chinese_cht繁體辨識問題

### DISCUSSION THREAD

**Discussion by aexss12378** [PaddlePaddle/PaddleOCR#13624] [Q&A] [answered]
'from paddleocr import PaddleOCR, draw_ocr

ocr = PaddleOCR(
    use_angle_cls=True, 
    lang="chinese_cht",
    det_model_dir='C:\\Users\\user/.paddleocr/whl\\det\\ch\\ch_PP-OCRv4_det_server_infer\\',
    rec_model_dir='C:\\Users\\user/.paddleocr/whl\\rec\\ch\\ch_PP-OCRv4_rec_server_infer',
    rec_char_dict_path='d:\\PaddleOCR_Project\\PaddleOCR-main\\ppocr\\utils\\dict\\chinese_cht_dict.txt',)
img_path = './doc/imgs/image.jpg'
result = ocr.ocr(img_path, cls=True)
for idx in range(len(result)):
    res = result[idx]
    for line in res:
        print(line)

from PIL import Image
result = result[0]
image = Image.open(img_path).convert('RGB')
boxes = [line[0] for line in result]
txts = [line[1][0] for line in result]
scores = [line[1][1] for line in result]
im_show = draw_ocr(image, boxes, txts, scores, font_path='./doc/fonts/1.ttf')
im_show = Image.fromarray(im_show)
im_show.save('result.jpg')'
辨識語言需求是繁體中文，但顯示亂碼
請問是代碼錯誤嗎
![result](https://github.com/user-attachments/assets/1816764e-1e6c-4ffc-92a9-6301513d4f91)

**Comment by SWHL** [comment]
你这问题太抽象了，可能是代码错误吧
建议提供完整运行环境和相关信息，保证我们可以复现

**Reply by aexss12378** [reply]
謝謝回覆!似乎是`ch_PP-OCRv4_rec_server_infer`不能用於辨識繁體中文（chinese_cht）
因此我改用`chinese_cht_PP-OCRv3_rec`
效果不是很好，請問這方面該如何改善呢

**Comment by GreatV** [comment]
`ch_PP-OCRv4_rec_server_infer` 是简体中文模型，加载繁体中文字典会乱码。

**Reply by aexss12378** [reply]
謝謝，但我基於這個QA[https://github.com/PaddlePaddle/PaddleOCR/discussions/13470#discussioncomment-10136543](url)
請問我應該使用的是`chinese_cht_PP-OCRv3_rec`這個模型嗎
但同樣回到辨識結果不佳的問題（ex.台灣辨識成合灣）
如圖
![result](https://github.com/user-attachments/assets/d8ffaf80-5c62-46c2-97ba-b529b8462744)

**以下是我使用的代碼：**
from paddleocr import PaddleOCR, draw_ocr

ocr = PaddleOCR(
    use_angle_cls=True, 
    lang="chinese_cht",
    det_model_dir='C:\\Users\\user/.paddleocr/whl\\det\\ch\\ch_PP-OCRv4_det_server_infer\\',
    rec_model_dir='C:\\Users\\user/.paddleocr/whl\\rec\\chinese_cht\\chinese_cht_PP-OCRv3_rec_infer',
    rec_char_dict_path='d:\\PaddleOCR_Project\\PaddleOCR-main\\ppocr\\utils\\dict\\chinese_cht_dict.txt'
    )
img_path = './doc/imgs/image.jpg'
result = ocr.ocr(img_path, cls=True)
for idx in range(len(result)):
    res = result[idx]
    for line in res:
        print(line)

from PIL import Image
result = result[0]
image = Image.open(img_path).convert('RGB')
boxes = [line[0] for line in result]
txts = [line[1][0] for line in result]
scores = [line[1][1] for line in result]
im_show = draw_ocr(image, boxes, txts, scores, font_path='./doc/fonts/1.ttf')
im_show = Image.fromarray(im_show)
im_show.save('result.jpg')

**環境：**

- OS:Windows
- cuda:12.0
- Python version: 3.8.19
- Paddle version: 2.6.1
- PaddleOCR:2.7.3

**相關訊息：**[2024/08/09 12:14:28] ppocr DEBUG: Namespace(alpha=1.0, alphacolor=(255, 255, 255), benchmark=False, beta=1.0, binarize=False, cls_batch_num=6, cls_image_shape='3, 48, 192', cls_model_dir='C:\\Users\\user/.paddleocr/whl\\cls\\ch_ppocr_mobile_v2.0_cls_infer', cls_thresh=0.9, cpu_threads=10, crop_res_save_dir='./output', det=True, det_algorithm='DB', det_box_type='quad', det_db_box_thresh=0.6, det_db_score_mode='fast', det_db_thresh=0.3, det_db_unclip_ratio=1.5, det_east_cover_thresh=0.1, det_east_nms_thresh=0.2, det_east_score_thresh=0.8, det_limit_side_len=960, det_limit_type='max', det_model_dir='C:\\Users\\user/.paddleocr/whl\\det\\ch\\ch_PP-OCRv4_det_server_infer\\', det_pse_box_thresh=0.85, det_pse_min_area=16, det_pse_scale=1, det_pse_thresh=0, det_sast_nms_thresh=0.2, det_sast_score_thresh=0.5, draw_img_save_dir='./inference_results', drop_score=0.5, e2e_algorithm='PGNet', e2e_char_dict_path='./ppocr/utils/ic15_dict.txt', e2e_limit_side_len=768, e2e_limit_type='max', e2e_model_dir=None, 
e2e_pgnet_mode='fast', e2e_pgnet_score_thresh=0.5, e2e_pgnet_valid_set='totaltext', enable_mkldnn=False, fourier_degree=5, gpu_id=0, gpu_mem=500, help='==SUPPRESS==', image_dir=None, image_orientation=False, invert=False, ir_optim=True, kie_algorithm='LayoutXLM', label_list=['0', '180'], lang='chinese_cht', layout=True, layout_dict_path=None, layout_model_dir=None, layout_nms_threshold=0.5, layout_score_threshold=0.5, max_batch_size=10, max_text_length=25, merge_no_span_structure=True, min_subgraph_size=15, mode='structure', ocr=True, ocr_order_method=None, ocr_version='PP-OCRv4', output='./output', page_num=0, precision='fp32', process_id=0, re_model_dir=None, rec=True, rec_algorithm='SVTR_LCNet', rec_batch_num=6, rec_char_dict_path='d:\\PaddleOCR_Project\\PaddleOCR-main\\ppocr\\utils\\dict\\chinese_cht_dict.txt', rec_image_inverse=True, rec_image_shape='3, 48, 320', rec_model_dir='C:\\Users\\user/.paddleocr/whl\\rec\\chinese_cht\\chinese_cht_PP-OCRv3_rec_infer', recovery=False, return_word_box=False, save_crop_res=False, save_log_path='./log_output/', savefile=False, scales=[8, 16, 32], ser_dict_path='../train_data/XFUND/class_list_xfun.txt', ser_model_dir=None, show_log=True, sr_batch_num=1, sr_image_shape='3, 32, 128', sr_model_dir=None, structure_version='PP-StructureV2', table=True, table_algorithm='TableAttn', table_char_dict_path=None, table_max_len=488, table_model_dir=None, total_process_num=1, type='ocr', use_angle_cls=True, use_dilation=False, use_gpu=True, use_mlu=False, use_mp=False, use_npu=False, use_onnx=False, use_pdf2docx_api=False, use_pdserving=False, use_space_char=True, use_tensorrt=False, use_visual_backbone=True, use_xpu=False, vis_font_path='./doc/fonts/simfang.ttf', warmup=False)

**Reply by GreatV** [reply, marked as answer]
如果使用繁体中文的模型还是效果不好的话，可能需要自己微调一下。
